core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -4901,7 +4901,7 @@ public void validateMerge(SqlMerge call) {
     if (call.getUpdateCall() != null) {
       validateUpdate(call.getUpdateCall());
     }
-    if (call.getInsertCall() != null) {
+    if (call.getInsertCall() != null && config.sqlConformance().allowMergeInsertRewrite()) {
```

Comment:
nit: line wrap

Reply:
Fix in https://github.com/googleinterns/calcite/pull/139/commits/20a224efcc552eb936868ade31f0f9883ea844f6